packages/taxdat/R/stan_input_helpers.R

```diff
@@ -175,6 +175,43 @@ reorder_single_source <- function(A,
               reordering = topleftorder[bfs_order]))
 }
 
+#' check_paralle_setup
+#'
+#' @param do_parallel 
+#' @param n_cpus 
+#'
+#' @export
+#'
+check_parallel_setup <- function(do_parallel, 
+                                 n_cpus) {
+  if(do_parallel) {
+    
+    library(foreach)
+    
+    if(n_cpus == 0)
+      stop("Specify the number of CPUS to use")
+    
+    if (!exists("cl")) {
+      cat("---- Initializing implicit cluster of", n_cpus, "cores \n")
+      # Parallel setup for foreach
+      doParallel::registerDoParallel(n_cpus)
+      cl <<- TRUE
+    }
+  } else {
+    cat("Parallel setup done \n")
+  }
+}
+
+#' close_parellel_setup
+#'
+#' @return
+#' @export
+#'
+#' @examples
+close_parellel_setup <- function() {
```

Comment:
rename to close_parallel_setup
I also didn't see anywhere where this function was called...